cmd/deploy/deploy.go

```diff
@@ -303,61 +324,48 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 			SkipIfExists: !deployOptions.Dependencies,
 		}
 		if err := pipelineCMD.ExecuteDeployPipeline(ctx, pipOpts); err != nil {
+			if errStatus := updateConfigMapStatus(ctx, cfg, c, data, err); err != nil {
+				return errStatus
+			}
+
 			return err
 		}
 	}
 
+	var errBuild error
 	if deployOptions.Build {
 		buildOptions := &types.BuildOptions{
 			EnableStages: true,
 			Manifest:     deployOptions.Manifest,
 			CommandArgs:  deployOptions.servicesToDeploy,
 		}
 		oktetoLog.Debug("force build from manifest definition")
-		if err := dc.Builder.Build(ctx, buildOptions); err != nil {
-			return err
-		}
+		errBuild = dc.Builder.Build(ctx, buildOptions)
 	} else {
-		svcsToBuild, err := dc.Builder.GetServicesToBuild(ctx, deployOptions.Manifest, deployOptions.servicesToDeploy)
-		if err != nil {
-			return err
-		}
-		if len(svcsToBuild) != 0 {
+		svcsToBuild, errBuild := dc.Builder.GetServicesToBuild(ctx, deployOptions.Manifest, deployOptions.servicesToDeploy)
+		if len(svcsToBuild) != 0 && errBuild != nil {
 			buildOptions := &types.BuildOptions{
 				CommandArgs:  svcsToBuild,
 				EnableStages: true,
 				Manifest:     deployOptions.Manifest,
 			}
-			if err := dc.Builder.Build(ctx, buildOptions); err != nil {
-				return err
-			}
+			errBuild = dc.Builder.Build(ctx, buildOptions)
```

Comment:
This `errBuild` refers to the one inside the else clausure, but we are checking it from outside the if else statement. Can you check it out pls